### PR TITLE
[improve][broker][PIP-149]Make GetMessageIdByTimestamp pure async

### PR DESCRIPTION
**Pull request by AnonHxy**
Master Issue: https://github.com/apache/pulsar/issues/14365

### Motivation

* See https://github.com/apache/pulsar/issues/14365

### Modifications

* Make `internalGetMessageIdByTimestamp` async

### Verifying this change

- [x] Make sure that the change passes the CI checks.

### Documentation

  
- [x] `doc-not-needed`

### CODE REVIEW

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java [hunk 1]
```diff
@@ -2740,53 +2740,50 @@ public void readEntryComplete(Entry entry, Object ctx) {
...
+        if (topicName.isGlobal()) {
+            future = validateGlobalNamespaceOwnershipAsync(namespaceName);
+        } else {
+            future = CompletableFuture.completedFuture(null);
+        }
 
-            if (!topicName.isPartitioned() && getPartitionedTopicMetadata(topicName,
-                    authoritative, false).partitions > 0) {
-                throw new RestException(Status.METHOD_NOT_ALLOWED,
+        return future.thenCompose(__ -> getPartitionedTopicMetadataAsync(topicName, authoritative, false))
```
Comment (Technoboy-) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java:2751]: If topicName.isPartitioned() = true, we don't need to `getPartitionedTopicMetadataAsync`.

Comment (AnonHxy) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java]: Nice catch. Updated, PTAL @Technoboy- @Jason918

**Review by Technoboy- — APPROVED** [review]

**Review by mattisonchao — APPROVED** [review]

**Pull request merged**